### ISSUE TITLE
Multi extruder

### ISSUE THREAD

**Issue by mdalacu** [SergiuToporjinschi/OctoPrint-CalibrationTools#17]
Hi, please make it possible to select different tools so we can calibrate our IDEX/multi extruders printers. 
Thx

**Comment by SergiuToporjinschi** [comment]
Hi @mdalacu,
Unfortunately, I have just a standard Ender 3 v2 with single extruder.
But what it needs to have multi extruder? 
I'm asking because in PID autotune you can change extruder index by incrementing E value.
How you are tuning steps for extruder?
And can you provide an example of response e for `M92`? because I think that should be different in case of multi extruder.
Thank you!

**Comment by mdalacu** [comment]
Hi!
Send: M92
Recv:   M92 X64.00 Y80.00 Z400.00
Recv:   M92 T0 E415.00
Recv: ok
[...]
Send: T1
Recv: echo:Active Extruder: 1
Recv: ok
[...]
Send: M92
Recv:   M92 X64.00 Y80.00 Z400.00
Recv:   M92 T1 E415.00
Recv: ok
T0 is selected as default....
thanks.    On Wednesday, February 16, 2022, 09:26:42 AM GMT+2, Sergiu Toporjinschi ***@***.***> wrote:  
 
 


Hi @mdalacu,
Unfortunately, I have just a standard Ender 3 v2 with single extruder.
But what it needs to have multi extruder?
I'm asking because in PID autotune you can change extruder index by incrementing E value.
How you are tuning steps for extruder?
And can you provide an example of response e for M92? because I think that should be different in case of multi extruder.
Thank you!

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by mdalacu** [comment]
So, basicly you need a new dropdown box with items T0...Tn , n beeing the number configured in Octoprint, printer profiles as "Number of Extruders". And on change...send Tn command before reading M92 or setting something.
Thanks

**Comment by SergiuToporjinschi** [comment]
oohh so that's completely different ...
I cannot promise anything yet... I working on this on my spare time

**Comment by SergiuToporjinschi** [comment]
> So, basicly you need a new dropdown box with items T0...Tn , n beeing the number configured in Octoprint, printer profiles as "Number of Extruders". And on change...send Tn command before reading M92 or setting something. Thanks

yes but is not so simple because I need the initial E value from M92 and there is another line involved in the printer response ...

**Comment by mdalacu** [comment]
Which one? The response contains only 2 linesRecv:   M92 X64.00 Y80.00 Z400.00
Recv:   M92 T0 E415.00
    On Wednesday, February 16, 2022, 11:30:08 AM GMT+2, Sergiu Toporjinschi ***@***.***> wrote:  
 
 



So, basicly you need a new dropdown box with items T0...Tn , n beeing the number configured in Octoprint, printer profiles as "Number of Extruders". And on change...send Tn command before reading M92 or setting something. Thanks


yes but is not so simple because I need the initial E value from M92 and there is another line involved in the printer response ...

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
